### PR TITLE
Add examples to docstrings of subgraph_(iso/monomorphism) methods

### PR DESCRIPTION
**Pull request by rossbar**
Addresses #7615 

Adds examples to the `subgraph_is_(iso/mono)morphic` and `subgraph_(iso/mono)morophism_iter` to highlight the role that argument order plays when constructing the GraphMatcher objects.

After some back and forth, I ultimately decided to do a separate docstring for each of the 8 methods. This means I added stubs for the `DiGraphMatcher` methods for the sole purpose of updating the docstrings. I had tried adding *both* a directed and undirected example to the `GraphMatcher` methods (which are then inherited and thus don't need to be duplicated), but decided against it for the subjective reason that I thought the examples section got too long when I did so. One solution would be to make shorter examples... I'm happy to give that a stab as well if it's preferred --- feedback welcome!

This PR also includes a few minor rst cleanups and comment removals.

### CODE REVIEW

File: networkx/algorithms/isomorphism/isomorphvf2.py [hunk 1]
```diff
@@ -846,6 +923,102 @@ def syntactic_feasibility(self, G1_node, G2_node):
...
+        >>> G = nx.DiGraph([(1, 2), (2, 1), (2, 3), (3, 2)])
+        >>> H = nx.DiGraph(nx.path_graph(5))
+
+        Check whether a subgraph of G is isomorphic to H:
+
+        >>> isomatcher = nx.isomorphism.DiGraphMatcher(G, H)
+        >>> isomatcher.subgraph_is_isomorphic()
+        False
+
+        Check whether as subgraph of H is isomorphic to G:
```
Comment (dschult) [inline networkx/algorithms/isomorphism/isomorphvf2.py:942]:
```suggestion
        Check whether a subgraph of H is isomorphic to G:
```

File: networkx/algorithms/isomorphism/isomorphvf2.py [hunk 2]
```diff
@@ -846,6 +923,102 @@ def syntactic_feasibility(self, G1_node, G2_node):
...
+        >>> G = nx.DiGraph([(0, 1), (2, 1), (3, 2)])
+        >>> H = nx.DiGraph([(0, 1), (1, 2), (2, 3), (3, 2)])
+
+        Check whether a subgraph of G is monomorphic to H:
+
+        >>> isomatcher = nx.isomorphism.DiGraphMatcher(G, H)
+        >>> isomatcher.subgraph_is_monomorphic()
+        False
+
+        Check whether as subgraph of H is isomorphic to G:
```
Comment (dschult) [inline networkx/algorithms/isomorphism/isomorphvf2.py:966]: looks like this typo is in each doc_string
```suggestion
        Check whether a subgraph of H is isomorphic to G:
```

File: networkx/algorithms/isomorphism/isomorphvf2.py [hunk 3]
```diff
@@ -846,6 +923,102 @@ def syntactic_feasibility(self, G1_node, G2_node):
...
+
+    def subgraph_isomorphisms_iter(self):
+        """Generator over isomorphisms between a subgraph of ``G1`` and ``G2``.
+
+        Examples
+        --------
+        When creating the `DiGraphMatcher`, the order of the arguments is important
+
+        >>> G = nx.DiGraph([(1, 2), (2, 1), (2, 3), (3, 2)])
+        >>> H = nx.DiGraph(nx.path_graph(5))
```
Comment (dschult) [inline networkx/algorithms/isomorphism/isomorphvf2.py:982]: The nodes here make it hard to see if a mapping is from H to G or from G to H.
Could we use `G = nx.DiGraph([("A", "B"), ("B", "C"), ("B", "A"), ("C", "B")])`?
Then the result is something like: `{0: 'A', 1: 'B', 2: 'C'}` 

Another choice could be: `G = nx.DiGraph([(101, 102), (102, 101), (102, 103), (103, 102)])`

Also, would it be simpler to make the graphs directed paths:
```python
G = nx.DiGraph([("A", "B"), ("B", "C")])
H = nx.DiGraph(nx.path_graph(5).edges)
```
But maybe that isn't simpler... it removes one direction from each node-pair. But the constructor syntax is longer.

**Review by dschult — COMMENTED** [review]
Nice!  
I found a typo. And I wonder if we can make the nodes more clearly part of `G` or `H` by avoiding small integers for one of the graphs. see below

File: networkx/algorithms/isomorphism/isomorphvf2.py [hunk 4]
```diff
@@ -846,6 +923,102 @@ def syntactic_feasibility(self, G1_node, G2_node):
...
+        >>> G = nx.DiGraph([("B", "C"), ("C", "B"), ("C", "D"), ("D", "C")])
+        >>> H = nx.DiGraph(nx.path_graph(5))
+
+        Yield isomorphic mappings between ``H`` and subgraphs of ``G``:
+
+        >>> isomatcher = nx.isomorphism.DiGraphMatcher(G, H)
+        >>> list(isomatcher.subgraph_isomorphisms_iter())
+        []
+
+        Check whether a subgraph of H is isomorphic to G:
```
Comment (dschult) [inline networkx/algorithms/isomorphism/isomorphvf2.py:990]:
```suggestion
        Yield isomorphic mappings between ``G`` and subgraphs of ``H``:
```

File: networkx/algorithms/isomorphism/isomorphvf2.py [hunk 5]
```diff
@@ -846,6 +923,102 @@ def syntactic_feasibility(self, G1_node, G2_node):
...
+        """Generator over monomorphisms between a subgraph of ``G1`` and ``G2``.
+
+        Examples
+        --------
+        When creating the `DiGraphMatcher`, the order of the arguments is important.
+
+        >>> G = nx.DiGraph([("A", "B"), ("C", "B"), ("D", "C")])
+        >>> H = nx.DiGraph([(0, 1), (1, 2), (2, 3), (3, 2)])
+
+        Check whether a subgraph of G is monomorphic to H:
```
Comment (dschult) [inline networkx/algorithms/isomorphism/isomorphvf2.py:1008]:
```suggestion
        Yield monomorphic mappings between ``H`` and subgraphs of ``G``:
```

File: networkx/algorithms/isomorphism/isomorphvf2.py [hunk 6]
```diff
@@ -846,6 +923,102 @@ def syntactic_feasibility(self, G1_node, G2_node):
...
+
+        >>> isomatcher = nx.isomorphism.DiGraphMatcher(G, H)
+        >>> list(isomatcher.subgraph_monomorphisms_iter())
+        []
+
+        Check whether a subgraph of H is isomorphic to G:
```
Comment (dschult) [inline networkx/algorithms/isomorphism/isomorphvf2.py:1014]:
```suggestion
        Yield monomorphic mappings between ``G`` and subgraphs of ``H``:
```

**Review by dschult — COMMENTED** [review]
I'm sorry that I didn't catch this before. But the last three intro phrases for examples are "Check whether...." when they should be  "Yield ... mappings ".

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**